### ISSUE TITLE
`influxd` installed by `influxdb` formula doesn't work as expected

### ISSUE THREAD

**Issue by danxmoran** [Homebrew/homebrew-core#75302]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
$ brew config
HOMEBREW_VERSION: 3.1.2
ORIGIN: https://github.com/Homebrew/brew
HEAD: bc9b98aa0ba777889218ac51f67ccc76b0be6a28
Last commit: 2 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 147fdabb3101aaedd59c22511f61f8a11ff12432
Core tap last commit: 53 seconds ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: dodeca-core 64-bit kabylake
Clang: 12.0 build 1200
Git: 2.31.1 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.2.3-x86_64
CLT: 12.4.0.0.1.1610135815
Xcode: N/A

$ brew doctor
Your system is ready to brew.
```

### 

- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

Users have been [reporting](https://github.com/influxdata/influxdb/issues/21133) in the `influxdb` repo that our [getting started guide](https://docs.influxdata.com/influxdb/v2.0/get-started/) doesn't work when installing `influxd` via Homebrew.

1. `brew install influxdb`
2. `influxd` to start the server

### What happened (include all command output)?

First, this error:
```
$ influxd
2021-04-16T13:25:39.901762Z     info    Welcome to InfluxDB     {"log_id": "0TYi_gkG000", "version": "2.0.4", "commit": "none", "build_date": "2021-04-16T13:25:35Z"}
2021-04-16T13:25:39.901941Z     error   Failed opening bolt     {"log_id": "0TYi_gkG000", "error": "unable to create directory /root/.influxdbv2/influxd.bolt: mkdir /root: read-only file system"}
Error: unable to create directory /root/.influxdbv2/influxd.bolt: mkdir /root: read-only file system
See 'influxd -h' for help
```

I can override the config to write outside of `/root`:
```
$ export INFLUXD_BOLT_PATH=~/.influxdbv2/influxd.bolt
$ export INFLUXD_ENGINE_PATH=~/.influxdbv2/engine
$ influxd
2021-04-16T13:29:06.680240Z     info    Welcome to InfluxDB     {"log_id": "0TYimJU0000", "version": "2.0.4", "commit": "none", "build_date": "2021-04-16T13:29:02Z"}
2021-04-16T13:29:06.777856Z     info    Resources opened        {"log_id": "0TYimJU0000", "service": "bolt", "path": "/Users/dan/.influxdbv2/influxd.bolt"}
2021-04-16T13:29:06.825796Z     info    Bringing up metadata migrations {"log_id": "0TYimJU0000", "service": "migrations", "migration_count": 14}
2021-04-16T13:29:11.075184Z     info    Using data dir  {"log_id": "0TYimJU0000", "service": "storage-engine", "service": "store", "path": "/Users/dan/.influxdbv2/engine/data"}
2021-04-16T13:29:11.075859Z     info    Compaction settings     {"log_id": "0TYimJU0000", "service": "storage-engine", "service": "store", "max_concurrent_compactions": 6, "throughput_bytes_per_second": 50331648, "throughput_bytes_per_second_burst": 50331648}
2021-04-16T13:29:11.075900Z     info    Open store (start)      {"log_id": "0TYimJU0000", "service": "storage-engine", "service": "store", "op_name": "tsdb_open", "op_event": "start"}
2021-04-16T13:29:11.076208Z     info    Open store (end)        {"log_id": "0TYimJU0000", "service": "storage-engine", "service": "store", "op_name": "tsdb_open", "op_event": "end", "op_elapsed": "0.310ms"}
2021-04-16T13:29:11.076926Z     info    Starting retention policy enforcement service   {"log_id": "0TYimJU0000", "service": "retention", "check_interval": "30m"}
2021-04-16T13:29:11.077306Z     info    Starting precreation service    {"log_id": "0TYimJU0000", "service": "shard-precreation", "check_interval": "10m", "advance_period": "30m"}
2021-04-16T13:29:11.078239Z     info    Starting query controller       {"log_id": "0TYimJU0000", "service": "storage-reads", "concurrency_quota": 10, "initial_memory_bytes_quota_per_query": 9223372036854775807, "memory_bytes_quota_per_query": 9223372036854775807, "max_memory_bytes": 0, "queue_size": 10}
2021-04-16T13:29:11.080795Z     info    Configuring InfluxQL statement executor (zeros indicate unlimited).     {"log_id": "0TYimJU0000", "max_select_point": 0, "max_select_series": 0, "max_select_buckets": 0}
2021-04-16T13:29:11.532488Z     info    Starting        {"log_id": "0TYimJU0000", "service": "telemetry", "interval": "8h"}
2021-04-16T13:29:11.532532Z     info    Listening       {"log_id": "0TYimJU0000", "transport": "http", "addr": ":8086", "port": 8086}
```

But then if I navigate to `http://localhost:8086`, the webpage returns:
```
500 Internal Server Error
```

### What did you expect to happen?

I expected:
1. The `influxd` binary installed via `brew` to be configured so that it doesn't try to write data into `/root` by default
2. The `influxd` binary installed via `brew` to successfully serve up UI assets

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew install influxdb
mkdir test-data
INFLUXD_BOLT_PATH=./test-data/influxd.bolt INFLUXD_ENGINE_PATH=./test-data/engine influxd &
# Wait for a log containing 'Listening...'
open http://localhost:8086
```

**Comment by danxmoran** [comment]
On the UI issue, the problem is that the `influxdb` formula is using `go build` to generate the binary, bypassing the project's `Makefile`. The Makefile contains config & logic to ensure that UI assets are properly bundled into the server binary (and to make sure the Rust pieces of the system are properly compiled and linked in, though that might be working as-is).

Changing the formula to run `make influx influxd` should help fix the problem.

**Comment by SMillerDev** [comment]
Please make a pull request if you think changing to `make` will help. But reading the makefile it doesn't actually seem to create any configuration so it doesn't solve your `/root/` issue.

**Comment by carlocab** [comment]
I actually did try using `make` at first when I was working on #73603, but it didn't work. I was using the instructions [here](https://github.com/influxdata/influxdb/blob/master/CONTRIBUTING.md#building-from-source).

Please do go ahead and open a pull request to fix it.

**Comment by danxmoran** [comment]
Sure, I can put something together.

> But reading the makefile it doesn't actually seem to create any configuration so it doesn't solve your /root/ issue.

That's true, I'll probably need to restore the `plist` that was removed in #73603 so I can point the server at config on startup.

> I actually did try using make at first when I was working on #73603, but it didn't work.

Do you remember what failures you saw?

**Comment by SMillerDev** [comment]
> That's true, I'll probably need to restore the plist that was removed in #73603 so I can point the server at config on startup.

That config doesn't exist for new users though, so we can't restore the plist. Unless influx starts shipping a default config there is nothing we can do since we don't make our own config files.

**Comment by carlocab** [comment]
>Do you remember what failures you saw?

It's been a while, so unfortunately not 😅 I think it had to do with the missing `pkg-config` wrapper, which I ended up fixing anyway. That plus I recall not being able to figure out how to pass the version in the `-ldflags` through the `Makefile`.

**Comment by sdennison-bv** [comment]
Was there any resolution for this issue? still seeing it

**Comment by danxmoran** [comment]
@sdennison-bv not yet, I had to wrap up some other work. Starting to work on a fix today.

**Comment by danxmoran** [comment]
> Unless influx starts shipping a default config there is nothing we can do since we don't make our own config files.

@SMillerDev `influxdb` doesn't have any checked-in default config files, but it does expose an `influxd print-config` command to generate a default config. Would it be OK to have the formula perform the equivalent of this shell logic?
```sh
if [ ! -f $(brew --prefix)/etc/influxdb2.yml ]; then
  influxd print-config --bolt-path $(brew --prefix)/var/influxdb2/influxd.bolt --engine-path $(brew --prefix)/var/influxdb2/engine > $(brew --prefix)/etc/influxdb2.yml
fi
```
And then configure the plist to point at `$(brew --prefix)/etc/influxdb2.yml`?

**Comment by danxmoran** [comment]
@carabolic I almost missed your edit of my comment above 😄 I'll submit a follow-up PR to implement the config-generation & attempt to restore the plist

**Comment by carlocab** [comment]
> @carabolic I almost missed your edit of my comment above 😄 I'll submit a follow-up PR to implement the config-generation & attempt to restore the plist

...oops. I definitely meant to quote your comment, not edit it. I am bad at GitHub.

**Comment by carlocab** [comment]
Fixed in #77754 and #77226.

Thanks, @danxmoran.
